test/functional/test_solution.py

```diff
@@ -0,0 +1,439 @@
+# Copyright 2106 F5 Networks Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import decorator
+import paramiko
+import pytest
+import time
+import sys
+
+from f5.bigip import BigIP
+from pprint import pprint as pp
+from neutronclient.v2_0 import client
+from f5_os_test.polling_clients import NeutronClientPollingManager
+
+
+lbaas_version = 1
+
+
+def log_test_call(func):
+    def wrapper(func, *args, **kwargs):
+        print("\nRunning %s" % func.func_name)
+        return func(*args, **kwargs)
+    return decorator.decorator(wrapper, func)
+
+
+class ExecTestEnv(object):
+    def __init__(self, symbols):
+        self.symbols = symbols.copy()
+        new_symbols = {
+            # leave the following settings alone
+            'bigip_username':           'admin',
+            'bigip_password':           'admin',
+            'lb_provider':              ('F5NetworksTest' if lbaas_version == 2
```

Comment:
Make sure we sync with @jlongstaf and @richbrowne.  I think we were going to remove the `Test` from this right?